### PR TITLE
MTDSA-705 savings

### PR DESCRIPTION
**Pull request by alexec**

### CODE REVIEW

File: public/api/conf/1.0/application.raml [hunk 1]
```diff
@@ -17,3 +22,24 @@ schemas:
         displayName: National Insurance Number
         type: string
         required: true
+    /bank-interest/{taxYear}:
```
Comment (alexec) [inline public/api/conf/1.0/application.raml]: Perhaps `total-bank-interest`?

**Comment by alexec** [comment]
As bank interest may be pre-populated later on, API needs to cater for it.

**Comment by alexec** [comment]
May be `/bank-accounts/total/incomes/{taxYear}`?

**Comment by saurabharora80** [comment]
Bank account Interest/Savings income needs to collected per bank-account, so we should also have 
POST | GET /nino/:nino/bank-accounts and PUT | GET /nino/:nino/bank-accounts/:accountId
followed by
POST | GET /nino/:nino/bank-accounts/:accountId/periods and PUT | GET /nino/:nino/bank-accounts/:accountId/periods/:periodId

File: public/api/conf/1.0/application.raml [hunk 2]
```diff
@@ -21,3 +24,23 @@ schemas:
...
+       uriParameters:
+         taxYear:
+           type: string
+           example: 2016-17
+       put:
+         description: Update annual summary
+         body:
+           application/json:
+             schema: SavingsAnnualSummary
+             example: !include examples/savings/AnnualSummary.json
```
Comment (alexec) [inline public/api/conf/1.0/application.raml]: Missing OK/error response.

File: public/api/conf/1.0/examples/savings/errors/Period_INVALID_REQUEST.json [hunk 3]
```diff
@@ -0,0 +1,16 @@
+{
+  "code": "INVALID_REQUEST",
+  "message": "Invalid request",
+  "errors": [
+    {
+      "code": "INVALID_MONETARY_AMOUNT",
+      "message": "taxedInterest should be non-negative number up to 2 decimal values",
+      "path": "/_embedded/incomes/taxedInterest"
```
Comment (alexec) [inline public/api/conf/1.0/examples/savings/errors/Period_INVALID_REQUEST.json]: Path incorrect.

File: public/api/conf/1.0/examples/savings/errors/Period_INVALID_REQUEST.json [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    },
+    {
+      "code": "INVALID_MONETARY_AMOUNT",
+      "message": "untaxedInterest should be non-negative number up to 2 decimal values",
+      "path": "/_embedded/expenses/untaxedInterest"
```
Comment (alexec) [inline public/api/conf/1.0/examples/savings/errors/Period_INVALID_REQUEST.json]: Path incorrect.

File: public/api/conf/1.0/schemas/savings/AnnualSummary.json [hunk 5]
```diff
@@ -0,0 +1,11 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "type": "object",
+  "title": "Annual Summary",
```
Comment (alexec) [inline public/api/conf/1.0/schemas/savings/AnnualSummary.json]: Should be title case

File: public/api/conf/1.0/application.raml (hunk 2 above)
Comment (alexec) [inline public/api/conf/1.0/application.raml]: Fixed

**Comment by saurabharora80** [comment]
same questions as dividends.

**Comment by alexec** [comment]
We plan to collect this on an account by account basis. Closing.

**Pull request closed without merging**